app/controllers/readings_controller.rb

```diff
@@ -16,7 +16,7 @@ def index
       @readings = @readings.where(toread: true)
       @page_subtitle = ts("Marked For Later")
     end
-    @readings = @readings.order("last_viewed DESC").page(params[:page])
+    @readings = @readings.left_joins(:work).merge(Work.visible_to_registered_user).or(Work.where(id: nil)).order("last_viewed DESC").page(params[:page])
```

Comment:
A join is required to query by work fields, with the `visible_to_registered_user` scope

It must be a left join and `.or(Work.where(id: nil))` included otherwise deleted works won't be included in the reading history.